components/Pricing.js

```diff
@@ -1,37 +1,41 @@
 import cn from 'classnames';
 import { useRouter } from 'next/router';
 import { useState } from 'react';
-import { postData } from '../utils/helpers';
-import { getStripe } from '../utils/initStripejs';
-import { useUser } from '../components/UserContext';
-import Button from './ui/Button';
+
+import Button from '@/components/ui/Button';
+import { postData } from '@/utils/helpers';
+import { getStripe } from '@/utils/stripe-client';
+import { useUser } from '@/utils/useUser';
 
 export default function Pricing({ products }) {
-  const [billingInterval, setBillingInterval] = useState('month');
-  const [loading, setLoading] = useState(false);
   const router = useRouter();
+  const [billingInterval, setBillingInterval] = useState('month');
+  const [priceIdLoading, setPriceIdLoading] = useState();
   const { session, userLoaded, subscription } = useUser();
 
   const handleCheckout = async (price) => {
-    setLoading(true);
+    setPriceIdLoading(price.id);
```

Comment:
Hm, this doesn't seem to work on preview. Also, it would be important to disable all buttons though.

Reply:
Good idea! I will fix that.